File: Leetcode/3423.go
package Leetcode

func maxAdjacentDistance(nums []int) int {
	ans := 0
	n := len(nums)
	for i := range n - 1 {
		ans = max(ans, abs(nums[i+1]-nums[i]))
	}
	return max(ans, abs(nums[n-1]-nums[0]))
}

func abs(a int) int {
	if a > 0 {
		return a
	} else {
		return -a
	}
}

File: Leetcode/3423.py
from typing import List


class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        return max(
            max(abs(nums[i + 1] - nums[i]) for i in range(len(nums) - 1)),
            abs(nums[0] - nums[-1]),
        )
